plugins/micromega/zify.ml

```diff
@@ -74,24 +79,22 @@ module HConstr = struct
 
   let iter = M.iter
 
+  let is_empty = M.is_empty
+
 end
 
+
 (** [get_projections_from_constant (evd,c) ]
     returns an array of constr [| a1,.. an|] such that [c] is defined as
     Definition c := mk a1 .. an with mk a constructor.
     ai is therefore either a type parameter or a projection.
  *)
 let get_projections_from_constant (evd, i) =
-  match Constr.kind  (unsafe_to_constr i) with
-  | Constr.Const (c, u) ->
-     (match Environ.constant_opt_value_in (Global.env ()) (c,u) with
-      | None -> failwith "Add Injection requires a constant (with a body)"
-      | Some c -> (
-        match EConstr.kind evd (EConstr.of_constr c) with
-        | App (c, a) -> Some a
-        | _ -> None ))
-  | _ -> None
-
+  match EConstr.kind evd (Tacred.hnf_constr (Global.env ()) evd i) with
```

Comment:
`hnf_constr` is notoriously slow. Any reason not to use standard weak head reduction here instead?

Reply:
BTW, what is the "standard" way ? Which function do you refer to exactly ?